src/Plugin.php

```diff
@@ -486,10 +492,28 @@ public function checkStates($scan_inactive_and_new_plugins = false, array $exclu
             }
         }
 
-       // Prevent duplicated checks
+        // Prevent duplicated checks
         $directories = array_unique($directories);
 
-       // Check all directories from the checklist
+        // Build plugin info data once before iterating to save performances
+        $plugins_informations = [];
+        $plugins_directories = new DirectoryIterator(GLPI_ROOT . '/plugins');
```

Comment:
Missing support for plugins added from the Marketplace?

Reply:
I'm not sure, the original code was only targeting /plugins too so maybe its on purpose ?